core/user_api.php

```diff
@@ -288,18 +291,20 @@ function user_is_email_unique( $p_email, $p_user_id = null ) {
 	}
 
 	$p_email = trim( $p_email );
+	// Escape SQL LIKE pattern chars to ensure exact match
+	$p_email = preg_replace( '/([%_])/', '\\\\$1', $p_email );
```

Comment:
The escaping for SQL LIKE should be part of the implementation for `$t_query->sql_ilike` function. In this PR it is done before each call.

Reply:
> The escaping for SQL LIKE should be part of the implementation for $t_query->sql_ilike function

No we don't want that, the replacement has to be performed by the caller. 

If we did it in DbQuery::sql_ilike(), then the function would no longer interpret LIKE wildcards `%` or `_`, effectively turning it into a case-insensitive `=` operator (which basically what I'm going after here).